web/src/main/java/org/openmrs/web/filter/update/UpdateFilter.java

```diff
@@ -334,67 +334,118 @@ public void checkLocaleAttributesForFirstTime(HttpServletRequest httpRequest) {
+					ResultSet results = null;
+					try {
+						results = statement.getResultSet();
+						if (results.next()) {
+							Integer userId = results.getInt(1);
+							DatabaseUpdater.setAuthenticatedUserId(userId);
+							String storedPassword = results.getString(2);
+							String salt = results.getString(3);
+							String passwordToHash = password + salt;
+							boolean result = Security.hashMatches(storedPassword, passwordToHash)
+							        && isSuperUser(connection, userId);
+							statement.close();
+							return result;
+						}
+					}
+					finally { // closing ResultSet Object
+						if (results != null) {
+							try {
+								results.close();
+							}
+							catch (Exception e1) {
+								log.warn("Failed to quietly close ResultSet", e1);
+							}
+						}
+					}
+				}
+			}
+			finally { // Closing Statement object
+				if (statement != null) {
+					try {
+						statement.close();
+					}
+					catch (Exception e2) {
+						log.warn("Failed to quietly close Statement", e2);
+					}
 				}
 			}
-			// Close statement
-			statement.close();
 		}
-		catch (Exception e) {
+		catch (Exception e3) {
 			log
 			        .error(
 			            "Error while trying to authenticate as super user. Ignore this if you are upgrading from OpenMRS 1.5 to 1.6",
-			            e);
+			            e3);
 			
 			// we may not have upgraded User to have retired instead of voided yet, so if the query above fails, we try
 			// again the old way
 			try {
 				String select = "select user_id, password, salt from users where (username = ? or system_id = ?) and voided = '0'";
-				PreparedStatement statement = connection.prepareStatement(select);
-				statement.setString(1, usernameOrSystemId);
-				statement.setString(2, usernameOrSystemId);
-				
-				if (statement.execute()) {
-					ResultSet results = statement.getResultSet();
-					if (results.next()) {
-						Integer userId = results.getInt(1);
-						DatabaseUpdater.setAuthenticatedUserId(userId);
-						String storedPassword = results.getString(2);
-						String salt = results.getString(3);
-						String passwordToHash = password + salt;
-						return Security.hashMatches(storedPassword, passwordToHash) && isSuperUser(connection, userId);
+				PreparedStatement statement = null;
+				try {
+					statement = connection.prepareStatement(select);
+					statement.setString(1, usernameOrSystemId);
+					statement.setString(2, usernameOrSystemId);
+					
+					if (statement.execute()) {
+						ResultSet results = null;
+						try {
+							results = statement.getResultSet();
+							if (results.next()) {
+								Integer userId = results.getInt(1);
+								DatabaseUpdater.setAuthenticatedUserId(userId);
+								String storedPassword = results.getString(2);
+								String salt = results.getString(3);
+								String passwordToHash = password + salt;
+								return Security.hashMatches(storedPassword, passwordToHash)
+								        && isSuperUser(connection, userId);
+							}

```

Comment:
You must add those extra close methods so that we can report error if close fails and not only warning from the finally block.